validator/sawtooth_validator/gossip/permission_verifier.py

```diff
@@ -363,3 +363,52 @@ def handle(self, connection_id, message_content):
 
         # if allowed pass message
         return HandlerResult(HandlerStatus.PASS)
+
+
+class IdentityCache():
+    def __init__(self, identity_view_factory, current_root_func):
+        self._identity_view_factory = identity_view_factory
+        self._identity_view = None
+        self._current_root_func = current_root_func
+        self._cache = {}
+
+    def __len__(self):
+        return len(self._cache)
+
+    def __contains__(self, item):
+        return item in self._cache
+
+    def __getitem__(self, item):
+        return self._cache.get(item)
+
+    def __iter__(self):
+        return iter(self._cache)
+
+    def get_role(self, item, state_root):
+        value = self._cache.get(item)
+        if value == -1 or value is None:
+            if self._identity_view is None:
+                self.update_view(state_root)
+            value = self._identity_view.get_role(item)
+            self._cache[item] = value
+        return value
+
+    def get_policy(self, item, state_root):
+        value = self._cache.get(item)
+        if value == -1 or value is None:
+            if self._identity_view is None:
+                self.update_view(state_root)
+            value = self._identity_view.get_policy(item)
+            self._cache[item] = value
+        return value
+
+    def forked(self):
+        for key in self._cache:
+            self._cache[key] = -1
```

Comment:
Could None be used here instead of -1? Or could you just delete the key from the dictionary?